Add cross Icon to test

rnukala1982 opened the pull request:
To create Cross Icon
<img width="1704" alt="Screenshot 2024-03-16 at 5 49 17 PM" src="https://github.com/writingmate/webapp/assets/64998012/1e3f15d0-1e33-4cae-859d-db55cc25c872">

vood commented:
This is great first start! Thank you. Would you mind to do small visual tweaks before merge. Currently the added X creates a lot of empty space on the right side, which is problematic on mobile screens where vertical space is very precious, to fix it put X inline with the Yearly/Monthly ToggleGroup. You can achieve this either using position absolute with top-X, right-X tailwind classes or by inserting X button in the same parent div with ToggleGroup and working with flex, justify-between tailwind classes. 

Thanks again for your work.

rnukala1982 commented:
Addressed the feedback. The ui looks as below post changes:

<img width="1765" alt="Screenshot 2024-03-17 at 10 17 58 AM" src="https://github.com/writingmate/webapp/assets/64998012/7234cfbd-9d3c-4909-9edf-61d10ad59501">

rnukala1982 commented:
Updated the PR to fix this. Thanks for catching this. Hope fully this matches the expectation.
![Uploading Screenshot 2024-03-17 at 11.57.20 AM.png…]()

vood commented on app/[locale]/setup/page.tsx:
```diff
@@ -210,7 +211,7 @@ export default function SetupPage() {
             showNextButton={true}
             showBackButton={true}
           >
-            <Plans />
```
I hate to do this, and it is my fault, I overlooked this nuance originally, but hey, an opportunity to learn the codebase :) 

So the `Plans` component is reusable and is shown in two places. One during the onboarding in `setup/page.tsx`, and the second time within the app itself in `plan-picker.tsx`.

We only need a cross icon in `plan-picker.tsx` and we do not need it in `setup/page.tsx`

You can achieve this two ways:
1. Introduce a `showCloseIcon` argument in Plans and making `onClose` optional
2. Moving the close icon code to plan-picker, this way you won't need to change `Plans` component at all

Thoughts?

Below the X showing in the onboarding, just for the reference (we don't need it there as I said earlier). 

<img width="758" alt="Screenshot 2024-03-17 at 4 00 49 PM" src="https://github.com/writingmate/webapp/assets/420428/a78022a3-e696-4297-ab3e-1a3d63dc5f76">

rnukala1982 replied:
Sure will check the options and fix it. I prefer option1 so if we need to reuse it anywhere we dont have to comeback and code again. Just flip the flag. I will check on changes in a little bit later today.

rnukala1982 replied:
Added the changes.
